great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -295,16 +342,244 @@ def run(
+                "expectation_configuration_builders": effective_expectation_configuration_builder_configs,
+            }
+        else:
+            effective_rule_config = rule_config
+
+        return effective_rule_config
+
+    @staticmethod
+    def _reconcile_domain_builder_config(
+        rule: Rule,
+        domain_builder_config: dict,
+    ) -> dict:
+        effective_domain_builder_config: dict = {}
+        batch_request: Optional[
+            Union[BatchRequest, RuntimeBatchRequest, dict]
+        ] = domain_builder_config.pop("batch_request", None)
+        if batch_request is None:
+            batch_request = get_batch_request_as_dict(
+                batch_request=rule.domain_builder.batch_request
+            )
+
+        effective_domain_builder_config["batch_request"] = batch_request
+
+        key: str
+        value: Any
+        current_value: Any
+        for key, value in domain_builder_config.items():
+            if hasattr(rule, f"{key}"):
+                current_value = getattr(rule, key)
+                effective_domain_builder_config[key] = value or current_value
+            else:
+                effective_domain_builder_config[key] = value
+
+        return effective_domain_builder_config
+
+    @staticmethod
+    def _reconcile_parameter_builder_configs_for_rule(
+        rule: Rule, rule_config: dict
+    ) -> Optional[List[dict]]:
+        effective_parameter_builder_configs: Optional[List[dict]] = []
+        parameter_builder_configs: Optional[List[dict]] = rule_config.get(
+            "parameter_builders", []
+        )
+
+        current_parameter_builders: Optional[
+            Dict[str, ParameterBuilder]
+        ] = rule.parameter_builders
+        parameter_builder_config: dict
+        for parameter_builder_config in parameter_builder_configs:
+            parameter_builder_name: str = parameter_builder_config["name"]
+            if parameter_builder_name in current_parameter_builders:
+                parameter_builder: ParameterBuilder = current_parameter_builders[
+                    parameter_builder_name
+                ]
+                effective_parameter_builder_configs.append(
+                    RuleBasedProfiler._reconcile_parameter_builder_config(
+                        parameter_builder=parameter_builder,
+                        parameter_builder_config=parameter_builder_config,
+                    )
+                )
+            else:
+                effective_parameter_builder_configs.append(parameter_builder_config)
+
+        return effective_parameter_builder_configs
+
+    @staticmethod
+    def _reconcile_parameter_builder_config(
+        parameter_builder: ParameterBuilder,
+        parameter_builder_config: dict,
+    ) -> dict:
+        effective_parameter_builder_config: dict = {}
+        batch_request: Optional[
+            Union[BatchRequest, RuntimeBatchRequest, dict]
+        ] = parameter_builder_config.pop("batch_request", None)
```

Comment:
This on the other hand is a true `Optional` value (because of the `None` default return value of `get`)

Reply:
@cdkini I fixed the above "optional" that became a list, so since it is not a true optional, I got rid of it.  Thanks.